test/tx_loc_modes2.ml

```diff
@@ -0,0 +1,32 @@
+(* Location is in Lock-free mode
+   Each Thread increments the value at a and reads a in a loop*)
+
+module Loc = Kcas.Loc
+module Mode = Kcas.Mode
+module Tx = Kcas.Tx
+
+let a = Loc.make ~mode:Mode.lock_free 4
+let read_loc loc = Tx.(return @@ Loc.get loc)
+let incr_loc loc v = Tx.(return @@ Loc.fetch_and_add loc v)
+let loop_count = 750
+let sleep_time = 0.01
+
+let thread1 () =
+  for _ = 1 to loop_count do
+    let alp = incr_loc a 1 in
+
+    let _ = Printf.printf "%d 1 i\n" @@ Tx.commit alp in
+
+    Unix.sleepf sleep_time
+  done
+
+let thread2 () =
+  for _ = 1 to loop_count do
```

Comment:
To make it more likely that the domains will be performing work simultaneously, you could use barrier synchronization.  Take a look at some of the other tests/benchmarks that use `Barrier` on how that can be done.

Reply:
Will modify as to  use Barriers instead of making domains sleep.